tests/rbd_mirror/test_rbd_mirror_image_migration.py

```diff
@@ -0,0 +1,262 @@
+
+Pre-requisites :
+Two ceph clusters with rbd mirror configured along with:
+        1. rbd mirror daemons per cluster
+        2. 1 client
+
+Test cases covered -
+CEPH-83573320 - Copy source image(Live migration) which is mirrored and check the behaviour
+
+Test Case Flow -
+1. Create an source EC pool with different k_m values and an image
+2. Create an EC pool for destination pool
+3. Mount the images and run io from clients
+4. Stop IO and prepare migration process from source pools to destination pool
+5. Start IO from client on destination images
+6. Execute migration
+7. Commit migration.
+8. repeate above steps(1-7) for Replication pool
+9. Cover test case for both pool based and image based journal mirroring
+"""
+
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import Rbd, verify_migration_commit, verify_migration_state
+from tests.rbd_mirror import rbd_mirror_utils as rbdmirror
+from tests.rbd_mirror.rbd_mirror_utils import rbd_mirror_config
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run_fio_and_migrate_image(rbd1, mirror1, image, src_spec, dest_pool, dest_spec, io):
+    # run io's on the source image
+    mirror1.benchwrite(imagespec=src_spec, io=io)
+
+    # Perpare the migration process
+    rbd1.migration_prepare(src_spec, dest_spec)
+
+    # verify migration prepare
+    verify_migration_state(rbd1, dest_spec)
+
+    # run io's on the destination image
+    mirror1.benchwrite(imagespec=dest_spec, io=io)
+
+    # Execute migration process
+    rbd1.migration_action(action="execute", dest_spec=dest_spec)
+
+    # verify migration execute
+    verify_migration_state(rbd1, dest_spec)
+
+    # commit migration process
+    rbd1.migration_action(action="commit", dest_spec=dest_spec)
+
+    # verify commit migration
+    verify_migration_commit(rbd1, dest_pool, image)
+
+
+def run(**kw):
+    """Image live migration for mirrored images.
+
+    Args:
+        kw: Key/value pairs of configuration information to be used in the test
+            Example::
+            config:
+                "ec-pool-only": True
+                ec_pool_config:
+                    pool: rbd_pool_4
+                    data_pool: rbd_ec_pool_4
+                    ec_profile: rbd_ec_profile_4
+                    image: rbd_image_4
+                    size: 10G
+
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+    """
+    log.info(
+        "Starting execution of test case :"
+        " CEPH-83573320 - Copy source image(Live migration) which is mirrored"
+    )
+    config = kw.get("config")
+    mirror1, mirror2 = [
+        rbdmirror.RbdMirror(cluster, config)
+        for cluster in kw.get("ceph_cluster_dict").values()
+    ]
+
+    rbd1, rbd2 = [
+        Rbd(**kw, req_cname=cluster_name)
+        for cluster_name in kw.get("ceph_cluster_dict").keys()
+    ]
+    for key, value in kw["config"].items():
+        if key == "pool_based":
```

Comment:
As we are executing test on both **pool based** and **image based** hence i feel we can pass these both in `for loop` to avoid code redundancy same with image based?

Reply:
done.